plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveBucketing.java

```diff
@@ -186,6 +186,10 @@ static int getBucketNumber(int hashCode, int bucketCount)
             return Optional.empty();
         }
 
+        if (!supportsBucketing(table)) {
+            return Optional.empty();
```

Comment:
>SHOW CREATE TABLE won't show bucketing (not desired, not sure if a bug deal)

I do not feel it is insanely important. Though I would prefer to have the bucketing information inf `SHOW CREATE TABLE` output if possible. On the other hand, we would not be able to execute such `CREATE TABLE ` statement via Trino. 
Not sure what is more important. Providing more information to user or have output of `SHOW ...` unable. I would opt for former.

Reply:
> (right? is the validation happening on CREATE?)

I'm afraid we don't validate it for timestamp or other data types. We fail when we try to compute the bucket number.

Reply:
@Praveen2112 @losipiuk and myself discussed this offline and decided that for a case of type that's not supported:

- SELECT should work
  - should not do validation
  - `$bucket` column should not be visible
- INSERT should fail
  - let’s make sure we have tests
- SHOW CREATE TABLE: should report all columns we bucket on, including unsupported ones
- CREATE TABLE should reject if user chooses to bucketed on unsupported column type
  - this is not the case today — today no validation
  - won’t be addressed in this PR
  - let’s make sure we have tests documenting current behavior
- CREATE TABLE AS SELECT should fail if user chooses to bucketed on unsupported column type
   - this is the case today, because hash function is not implemented
   - let’s make sure we have tests